### PR TITLE
RDoc-2384_resharding

### PR DESCRIPTION
**Pull request by reebhub**

### CODE REVIEW

**Review by karmeli87 — CHANGES_REQUESTED** [review]

File: Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown [hunk 1]
```diff
@@ -3,24 +3,267 @@
...
+Studio will indicate its progress in resharding the requested 
+buckets range using popup messages until the process ends.  
+
+!["Finished Resharding"](images/resharding_finished-resharding.png "Finished Resharding")
+
+---
+
+### Bucket Ownership
+
+For a while, as long as there are still files to transfer to the new
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown:238]: done

File: Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown [hunk 2]
```diff
@@ -3,24 +3,267 @@
...
+
+During [external replication](../sharding/external-replication), 
+we check **only the version part** of the document's change vector.  
+
+If a replicated document exists on the destination side:  
+To decide whether to replicate the document, we compare the source 
+document's **version** (right) part of the change vector with the 
+change vector of the destination document.  
+
+The **Order** part of the change vector is not used.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown:159]: During [external replication](../sharding/external-replication):  

* The **order** (left-hand) part of the document's change vector is checked on the **source** side, to determine whether the document should be replicated.  
* The **version** (right-hand) part of the document's change vector is checked on the **destination** side, to determine whether this version of the document already resides on it.

File: Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown [hunk 3]
```diff
@@ -3,24 +3,267 @@
...
+to replicate, for example) and its **order** on the database (so we can tell, 
+for example, whether to replicate or skip it).  
+
+On a sharded database, the latter (order) property may turn meaningless, because 
+resharding may change the order of documents: an old document may be moved to a shard 
+that contains newer documents, and get a change vector newer than theirs.  
+
+To overcome this problem, resharded documents receive an altered change vector 
+that explicitly defines both its version and its order, using this format:  
+`<document name> <order>|<version>`
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown:136]: To resolve this issue, resharded documents are given an altered change vector that explicitly defines both their version and their order, using this format:  
`<order>|<version>`

* E.g. `Users/1 A:3|B:7`  
  In the example above `A:3|B:7` is `Users/1`'s change vector.  
    * `A:3` indicates the document's **order**.  
    * The `|` symbol separates the two parts.  
    * `B:7` inidcates the document's **version**.

File: Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown [hunk 4]
```diff
@@ -3,24 +3,267 @@
...
+routinely checks the system.  
+Upon locating a file in a bucket that is already owned by another shard, 
+the documents migrator task immediately initiates a new resharding process 
+for the related bucket.  
+
+{PANEL/}
+
+{PANEL: Change Vector on a Sharded Database}
+
+On a non-sharded database, a document's [change vector](../server/clustering/replication/change-vector)
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown:125]: done

File: Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown [hunk 5]
```diff
@@ -3,24 +3,267 @@
...
+    * Shard #2 replies: `Ownership Transferred`.  
+      At this point, the transfer of the next bucket can start since 
+      no files are being transferred.  
+    * All bucket content is deleted from all replicas of shard #0.  
+      The bucket is **not** remapped to shard #2 until all of its files 
+      have been removed from the original shard. Until then, new content 
+      (document modification, new revision, time series update..) continues 
+      to be stored in the original shard and bucket. 
+      When the original bucket is cleared, the bucket is remapped 
+      to shard #2 and document modifications start to be routed there.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/resharding.markdown:79]: done

File: Documentation/6.0/Raven.Documentation.Pages/sharding/overview.markdown [hunk 6]
```diff
@@ -124,6 +124,17 @@ that would seamlessly retrieve `Users/3000` from shard **2** and `Users/5000` fr
 shard **3** and hand them to the client.  
 {NOTE/}
 
+As much as clients are concerned a sharded database is still a single entity: 
+the clients are not required to detect whether the database is sharded or not, 
+and clients of RavenDB versions prior to 6.0 (that had no sharding support) 
+can access a sharded database unaltered.  
+
+Shard-specific operations are, however, available: clients can, for example,
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/sharding/overview.markdown:132]: --> 
"Shard-specific operations are, however, available: a client can, for example, track the shard that a document is stored at and query this shard, and Studio can be used to relocate ([reshard](../sharding/resharding)) documents from one shard to another. "

**Pull request merged**